File: Pythonleetcode/Leetcode1365.py
class Solution(object):
    def smallerNumbersThanCurrent(self, nums):
        nums_sort=sorted(nums)
        prev=nums_sort[0]
        num_cnt=0
        nums_dict = {prev: num_cnt}
        for i in range(1,len(nums_sort)):
            num_cnt += 1
            if nums_sort[i]!=prev:
                prev = nums_sort[i]
                nums_dict[prev]= num_cnt
        ans = []
        for i in range(len(nums)):
            ans.append(nums_dict[nums[i]])
        return ans





print(Solution().smallerNumbersThanCurrent([8,1,2,2,3]))
print(Solution().smallerNumbersThanCurrent([6,5,4,8]))
print(Solution().smallerNumbersThanCurrent([7,7,7,7]))

File: CppPractice/Project1/Project1/Leetcode1365.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

using namespace std;


class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        map<int, int> dict;
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++)
        {
            if (dict.find(nums[i]) == dict.end())
            {
                dict[nums[i]] = 0;
            }
            dict[nums[i]] += 1;

        }
        int temp = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            for (auto pair : dict)
            {
                if (nums[i] > pair.first)
                {
                    temp += dict[pair.first];
                }
            }
            ans.push_back(temp);
            temp = 0;
        }
        return ans;

    }
};